ample/util/reference_manager.py

```diff
@@ -0,0 +1,265 @@
+            elif section == self.SECTIONS.MODELLING:
+                labels = []
+                if optd.get('ideal_helices'):
+                    labels.append('AMPLE_COILED-COILS')
+                if optd.get('make_models'):
+                    labels.append('ROSETTA')
+                if optd.get('nmr_model_in'):
+                    labels.append('AMPLE_NMR')
+                if optd.get('quark_models'):
+                    labels.append('QUARK')
+                    labels.append('AMPLE_QUARK')
+                if optd.get('transmembrane'):
+                    labels.append('AMPLE_TRANSMEMBRANE')
+                self.section_labels[section] = labels
+            elif section == self.SECTIONS.MODEL_PREP:
+                labels = []
+                if not optd.get('import_ensembles'):
+                    labels += ['CCTBX', 'THESEUS', 'GESAMT']
+                    if optd.get('use_scwrl'):
+                        labels.append('SCWRL4')
+                    elif optd['cluster_method'] in [SPICKER_RMSD, SPICKER_TM]:
+                        labels.append('SPICKER')
+                    elif optd['cluster_method'] in ['fast_protein_cluster']:
+                        labels.append('FPC')
+                    self.section_labels[section] = labels
+            if optd.get('do_mr'):
+                if section == self.SECTIONS.MR:
+                    labels = ['MRBUMP']
+                    if optd.get('mrbump_programs'):
+                        if 'molrep' in optd['mrbump_programs']:
+                            labels.append('MOLREP')
+                        if 'phaser' in optd['mrbump_programs']:
+                            labels.append('PHASER')
+                    self.section_labels[section] = labels
+                elif section == self.SECTIONS.REFINEMENT:
+                    self.section_labels[self.SECTIONS.REFINEMENT] = ['REFMAC']
+                elif section == self.SECTIONS.DM:
+                    if optd.get('use_shelxe'):
+                        self.section_labels[self.SECTIONS.DM].append('SHELXE')
+                elif section == self.SECTIONS.AUTOBUILD:
+                    labels = []
+                    if optd.get('refine_rebuild_arpwarp') or optd.get('shelxe_rebuild_arpwarp'):
+                        labels += ['ARPWARP']
+                    elif optd.get('refine_rebuild_buccaneer') or optd.get('shelxe_rebuild_buccaneer'):
+                        labels += ['BUCCANEER']        
+                    self.section_labels[section] = labels
+                
+        # Generate ordered list of all relevant reference labels
+        for section in self.SECTIONS:
+            if section in self.section_labels:
+                self.ordered_labels += self.section_labels[section]
+        return
+    
+    @property
+    def methods_as_html(self):
+        html = "<p>This section lists the programs and algorithms that were used in this job and the references that should be cited. " + \
```

Comment:
You could use a little formatting trick here. Instead of all the `<p>` and `</p>` tags, maybe add `{tags_open}` and `{tags_close}` to each string, and then use `str.format(tags_open='<p>', tags_close='</p>')`. It would allow us to manipulate later on much easier.